### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by jayj10080**
- Valid move for King, with tests.
- Capture logic, with tests.
- Removed hook in Game model, moved populate board method into controller under create action.

### CODE REVIEW

File: app/models/king.rb [hunk 1]
```diff
@@ -1,2 +1,24 @@
 class King < ChessPiece
+  belongs_to :game
```
Comment (hanshank) [inline app/models/king.rb]: Do we need this, since King is inheriting from ChessPiece?

File: app/models/king.rb [hunk 2]
```diff
@@ -1,2 +1,24 @@
...
+
```
Comment (hanshank) [inline app/models/king.rb]: Niece piece of code in here! Good job 👍

File: app/controllers/games_controller.rb [hunk 3]
```diff
@@ -36,6 +36,8 @@ def new
   def create
     @game = current_user.games.create(game_params)
     if @game.valid?
+      @game.populate_board
```
Comment (zacjones93) [inline app/controllers/games_controller.rb]: does this solve the double population issue we were running into?

File: app/models/king.rb (hunk 1 above)
Comment (zacjones93) [reply app/models/king.rb]: Hmm, that's a good question

File: app/controllers/games_controller.rb (hunk 3 above)
Comment (jayj10080) [reply app/controllers/games_controller.rb]: I believe so.

File: app/models/king.rb (hunk 1 above)
Comment (jayj10080) [reply app/models/king.rb]: My tests failed unless I did this.

File: app/controllers/games_controller.rb (hunk 3 above)
Comment (mahoneyj02) [reply app/controllers/games_controller.rb]: Yep, I believe it would too.

**Pull request merged**